### PR TITLE
Tufnel/retire from cm plumbing 1106 refactor

### LINKED ISSUE #1097: [retirements] Retire from portfolio - initial plumbing
While waiting for finalized mocks we can start with the plumbing. This is high priority for this epic / milestone: https://github.com/KlimaDAO/klimadao/milestone/9

# Summary
Goal is to support retirement of your own assets, in carbonmark, without bouncing the user to klimadao.

This is the journey:
1. A user has already purchased a carbon asset, which can be seen listed in their /portfolio page.
2. They click RETIRE and are taken to an offset form.
3. They complete the form, with constraints and validation (follow-up PR)
4. They complete an approval txn, and a retirement txn, in the Transaction Modal.
5. On success, they are taken to a 'retirement successful' screen.
6. From that screen (after a few seconds delay for subgraph to catch up) they can click to navigate to their 'retirement page'.

# Engineering Requirements
For this PR we only want to write up the key data fetching, transaction logic, and navigation logic.
We don't have to finalize form validation or designs yet.
* [ ] Do *not* change the URL on the RETIRE button.
  - This allows us to merge this PR without exposing it to users until it's ready
image
* [ ] Set up a page to hold the form under the url `/porfolio/retire/[project_token_0x]`
  * [ ] When this page loads, if the user is not connected, bounce them back to /portfolio
  * [ ] When the page loads, fetch the relevant project info from subgraph. See dapp for an example of how we do that.|
    - project token balance, project name, project id, vintage, project category.
* [ ] Write retirement functions that use aggregator v2, feel free to copy paste as much as you can from the existing logic.
  * [ ] Existing logic can be simplified now, because it's only concerned with retiring an asset you hold (no payment methods, prices etc.)

### PR DESCRIPTION
**Pull request by psparacino**
## Description

**This is a refactor of PR #1137** 

The retire link from the portfolio is not hooked up. To get to the retire page and test: `/portfolio/<token address here>/retire`

As `<Transaction />` is now being re-used, it now has an extra prop for the `spenderAddress`. I added it in the other components that needed it.

Given that the shared Modal is being used there are some style differences from the figma design. It might be worth adding some additional optional props to make the modal more customizable for different uses.

## Related Ticket

Resolves #1097 #1137

## Changes from Issue Requirements

**URL format**
Screen Shot 2023-05-30 at 8 26 46 PM

## Checklist

- [X] I have run `npm run build-all` without errors
- [X] I have formatted JS and TS files with `npm run format-all`

### CODE REVIEW

File: carbonmark/components/Transaction/Approve.tsx
Comment (ladytrekker) [inline carbonmark/components/Transaction/Approve.tsx:70]: How come that the price is being removed here?
Can you point to the figma designs for that?

This Transaction Modal is used here:
- Create Listing
- Edit Listing
- Purchase a Listing

Editing a Listing always comes with the price.

File: carbonmark/components/Transaction/index.tsx
Comment (ladytrekker) [inline carbonmark/components/Transaction/index.tsx:84]: This Go Back Button was never included in any designs.
Still, wenn a transaction fails and the error message appears, the user should be able to go back to the former modal (not the transaction modal) to change the input values again.
This is the case for Editing a Listing for example whose inputs are also appearing in a modal (modal after modal). 

Just to be sure that this is still working.

File: carbonmark/components/pages/Retire/RetireModal/styles.ts
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireModal/styles.ts:29]: Just wondering about the font-style being addressed here directly.
We do have a typo-system for each variant of font.
For text there is the `Text` component which should be able to render the correct fonts.
Is this here not enough?

File: carbonmark/components/Transaction/Approve.tsx
Comment (ladytrekker) [reply carbonmark/components/Transaction/Approve.tsx:70]: | Before WITH PRICE | After WITHOUT PRICE  |
|---------|--------|
| Bildschirmfoto 2023-05-19 um 14 37 53  | Bildschirmfoto 2023-05-19 um 14 37 53
 |

**Review by ladytrekker — COMMENTED** [review]
Hey @psparacino 

Having a quick glance at this PR as I am working on retirement from Project details (and may use your code here and there)

See some comments below.

Also I was not able to continue with the retirement on the preview link.
You may be aware of that?

https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/portfolio/0xd6ed6fae5b6535cae8d92f40f5ff653db807a4ea/retire
Bildschirmfoto 2023-05-19 um 14 49 55

File: carbonmark/lib/listingsGetter.ts
Comment (ladytrekker) [inline carbonmark/lib/listingsGetter.ts:86]: I would move this helper to /lib/createURLs

File: carbonmark/components/Transaction/Approve.tsx
Comment (psparacino) [reply carbonmark/components/Transaction/Approve.tsx:70]: > Hey @psparacino
> 
> Having a quick glance at this PR as I am working on retirement from Project details (and may use your code here and there)
> 
> See some comments below.
> 
> Also I was not able to continue with the retirement on the preview link. You may be aware of that?
> 
> https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/portfolio/0xd6ed6fae5b6535cae8d92f40f5ff653db807a4ea/retire Bildschirmfoto 2023-05-19 um 14 49 55

Thanks for taking a look; addressing comments individually below.

Re: retirement on the preview link, might be two reasons:

1) I was testing with a TC02 token; looks like you're using a C3T so I'll go back and confirm there are icons for all the carbon tokens (which looks like the reason for this failure).

2) When I tried using a TC02 token on the preview I still got a failure with `token` being undefined so maybe it has to do something with the env(?) in vercel? During development I had to specify the export `const DEFAULT_NETWORK = "mainnet";` otherwise it would default to testnet

**Comment by psparacino** [comment]
> Hey @psparacino
> 
> Having a quick glance at this PR as I am working on retirement from Project details (and may use your code here and there)
> 
> See some comments below.
> 
> Also I was not able to continue with the retirement on the preview link. You may be aware of that?
> 
> https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/portfolio/0xd6ed6fae5b6535cae8d92f40f5ff653db807a4ea/retire Bildschirmfoto 2023-05-19 um 14 49 55

Thanks for taking a look; addressing comments individually below.

Re: retirement on the preview link, might be two reasons:

I was testing with a TC02 token; looks like you're using a C3T so I'll go back and confirm there are icons for all the carbon tokens (which looks like the reason for this failure).

When I tried using a TC02 token on the preview I still got a failure with token being undefined so maybe it has to do something with the env(?) in vercel? During development I had to specify the export const DEFAULT_NETWORK = "mainnet"; otherwise it would default to testnet

File: carbonmark/components/Transaction/index.tsx
Comment (psparacino) [reply carbonmark/components/Transaction/index.tsx:84]: This was the design I had with the 'Go Back' Button:
Screen Shot 2023-05-19 at 10 19 32 AM

I had moved it up to the Retire Modal but I'll move it back to Transaction

File: carbonmark/components/pages/Retire/RetireModal/styles.ts
Comment (psparacino) [reply carbonmark/components/pages/Retire/RetireModal/styles.ts:29]: Will check that

**Comment by psparacino** [comment]
@ladytrekker Just summarizing my comments above:

Changes:
1) Moving Go Back Button back down to Transaction
2) optional prop in Transaction to show/hide price
3) Confirming all icons are accounted for
4) Font
5) Move link to helpers

I'll push these changes here.

The vercel failure might be a vercel settings issue, but will check again after these changes.

**Comment by jabby09** [comment]
@ShimonD-KlimaDAO this user flow is ready for initial QA

Note there was a related "plumbing" ticket (#1097) and the PR for all these changes is #1145 

PR link address to test from: https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/

Other testing notes:
- Currently the RETIRE button link is not hooked up (this is part of the plumbing work in the PR)
- You can just input the url to test the retire flow, i.e. `/portfolio/<token address here>/retire`
- To get the token address you can right click copy the link destination on the pop up notification redirect modal and the token address will be in there

**Review by jabby09 — CHANGES_REQUESTED** [review]
@psparacino (cc @ShimonD-KlimaDAO) initial testing feedback...

- [x] Default banner images are not using the darkened version so the white text is not readable. See a [project details](https://www.carbonmark.com/projects/VCS-1140-2015) page to see examples of the darkened default images. 
Screenshot 2023-05-23 at 9 42 25 AM

- [x] This copy does not match design, please change to:
> Be careful not to include any sensitive personal information (such as an email address) in your retirement name or message. The information you enter here cannot be edited once it is submitted and will permanently exist on a public blockchain.
Screenshot 2023-05-23 at 9 43 00 AM

- [x] Only the amount of tonnes is a required field so it needs the * label as per the design
Screenshot 2023-05-23 at 12 08 07 PM

- [x] When tonnes are entered, the RETIRE CARBON button should be activated
Screenshot 2023-05-23 at 9 44 05 AM

- [x] Beneficiary and retirement message are not required fields. Currently I had to enter these to get the RETIRE CARBON button to activate
Screenshot 2023-05-23 at 9 44 49 AM

- [x] As you noted in discord, Confirm Retirement modal is not following the design implementation in other parts of the site, this needs to change to be consistent

- [x] Please change the copy on the APPROVE screen to:

> You are about to retire a carbon asset.
> 
> The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.
> 
> Verify all information is correct and click 'approve' to continue.
Screenshot 2023-05-23 at 11 53 29 AM

- [x] Please change the copy of the SUBMIT screen to:

> The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.
> 
> To finalize your retirement, verify all information is correct and then click 'submit' below.
Screenshot 2023-05-23 at 11 54 40 AM

- [x] Processing retirement spinner modal is not matching the design dimensions, it appears much longer width
Screenshot 2023-05-23 at 9 47 06 AM

- [x] View transaction on polygonscan is not linking to the polygonscan transaction, it is linking to the retirement receipt
Screenshot 2023-05-23 at 12 00 39 PM

**Comment by ShimonD-KlimaDAO** [comment]
1. According to #1097 , the link should be `/porfolio/retire/[project_token_0x]`, not `/porfolio/[project_token_0x]/retire`.

2. The approval tab on the transaction modal has this:
image

3. The 'Submit' tab is explanation is not accurate. The approval does not grant access to move assets to the CM contracts (as with listings), but to the RA contracts.

4. Is "Balance (available to retire)" supposed to have the number of tonnes owned by the wallet? If not, I think this is a bit confusing at first glance. It makes it seem as if that's the number of tonnes the wallet owner can retire. 
image

5. Our capitalization of PolygonScan is inconsistent.

Version a (success modal in CM): 
image

Version b (retirements page): 
image

Version c (Asset detail pane in this PR):
image

6. I [tried](https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/portfolio/0x2f800db0fdb5223b3c3f354886d907a671414a7f/retire) putting in the BCT address to see what happens, and got a 404 error instead of bouncing me back to the Portfolio page as it does with tco2s and c3ts I don't own.

7. Some of the fields are off-center in mobile. There isn't a way to scroll right - it's just misaligned.
image

File: carbonmark/components/Transaction/HighlightValue.tsx
Comment (ladytrekker) [inline carbonmark/components/Transaction/HighlightValue.tsx:20]: Why do you name it here RetireProps?

File: carbonmark/components/Transaction/Submit.tsx
Comment (ladytrekker) [inline carbonmark/components/Transaction/Submit.tsx:62]: In which case can `value` be undefined?

File: carbonmark/components/Transaction/Submit.tsx
Comment (ladytrekker) [inline carbonmark/components/Transaction/Submit.tsx:68]: Here again => price is removed but still needed in other Transactions

File: carbonmark/components/Transaction/index.tsx
Comment (ladytrekker) [inline carbonmark/components/Transaction/index.tsx:27]: You can remove this prop as mentioned above.
Either a `price` is passed or not.

Also you are not using `showPrice` further down.

File: carbonmark/components/Transaction/index.tsx
Comment (ladytrekker) [inline carbonmark/components/Transaction/index.tsx:39]: I am wondering why we need an external `onViewChange` function?

The component has the prop `hasApproval` 
=> if this is `true` => show "Submit" view, else show "Approve"

If a Transaction has an approval should be passed down by the corresponding view (Edit Listing, Create Listing, Purchase a Listing, and here => Retire Modal)

**Review by ladytrekker — CHANGES_REQUESTED** [review]
Hey @psparacino 

Thanks for the changes 🌟 
I was able to retire ✅ 

In the future, I would prefer that all the bigger changes which touch the Transaction Modal are happening in a separated PR (before or after this PR).
Now we need to ensure that all Transactions are still working.
Right now, Creating A Listing throws on this PR for example (see comments below).

I did not take a look at the implementation of the actual retirement transaction yet.
Just ping me again for a final review! Thanks 🙏

File: carbonmark/components/CreateListing/index.tsx
Comment (psparacino) [reply carbonmark/components/CreateListing/index.tsx:210]: looks like it was was same name, ethers import instead of the klimadao util. fixed.

File: carbonmark/components/Transaction/HighlightValue.tsx
Comment (psparacino) [reply carbonmark/components/Transaction/HighlightValue.tsx:20]: relic from a previous version, removed.

File: carbonmark/components/Transaction/Approve.tsx
Comment (psparacino) [reply carbonmark/components/Transaction/Approve.tsx:70]: Yes, price should not be removed on create/edit, so fixing that. it does look the design change is across the board though for all transactions; this is the create listing design in figma:
create design

Create Approve:
create approve design

File: carbonmark/components/pages/Retire/RetireForm/index.tsx
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireForm/index.tsx:250]: There are already components for `Input` and `TextArea`. 
No need to write them again. 
See [here for example](https://github.com/KlimaDAO/klimadao/blob/staging/carbonmark/components/pages/Project/Purchase/PurchaseInputs.tsx#L59-L94)

Or what is the reason you did not use them?

File: carbonmark/components/pages/Retire/RetireForm/styles.ts
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireForm/styles.ts:214]: As mentioned above, there are already components for inputs etc.

File: carbonmark/pages/portfolio/[project_token_0x]/retire/index.tsx
Comment (ladytrekker) [inline carbonmark/pages/portfolio/[project_token_0x]/retire/index.tsx:53]: Why `undefined`? Or rather remove it from here as you catch it further down?

File: lib/constants/index.ts
Comment (ladytrekker) [inline lib/constants/index.ts:279]: Comment is in wrong place

**Review by ladytrekker — COMMENTED** [review]
Hey @psparacino 

I checked the code again.
Please see my comments below 🙏

**Comment by jabby09** [comment]
@psparacino for this PR preview link, can we please have the RETIRE button hooked up to the correct path, it is a bit cumbersome testing to have to copy the token address from the Leaving Carbonmark modal.

Screenshot 2023-05-26 at 9 27 44 AM

**Review by jabby09 — CHANGES_REQUESTED** [review]
@psparacino did another pass and identified more issues.

- [ ] Header image is not correct, the default image needs to be based on the Category. In this example, this is a Renewable Energy project but the image and Category in the retire form is "Other". Also the Verra (registry) pill is not rendering properly.

Screenshot 2023-05-26 at 9 30 14 AM

- [ ] Header image not matching design. On the retire screen, the image is not the correct proportions. See the image below from the design, the height of the banner image pushes the bottom below the bottom of the Asset Details component. Also note the rounded corners at the top of the form/image.

Screenshot 2023-05-26 at 10 24 14 AM

- [ ] The view transaction on PolygonScan link is not formatting correctly. It should be a link to the transaction, not the 0x address of the token. 

- For example, I did a retirement and this was the formatted link which results in a 404 error on polygonscan:
https://polygonscan.com/0x05e917686251e427034251087602da609e57f693/10/en

- The correct formatted link for that transaction I completed should be:
https://polygonscan.com/tx/0x3b7195d06acb7a082339e21b218718310da046ed3ef4b079c3a315d463b75948

Screenshot 2023-05-26 at 9 35 48 AM

- [x] Please make adjustment to Asset Details design.

- Note the label change to `Available to retire` the value of which should be the tonnes available in this users wallet.
- Note that View on PolygonScan is moved to the bottom of the component.

Screenshot 2023-05-26 at 12 01 48 PM

**Comment by psparacino** [comment]
@jabby09 @ladytrekker 

Everything should be ready for a final review. 

@jabby09: To address your request about linking the retire button for convenience and Atmos's original request to not link so it can be merged without redirecting yet, I just made a temporary retire button next to the regular one. If all looks good on the review let me know and I'll take it out for the merge.

**Comment by jabby09** [comment]
@ShimonD-KlimaDAO not sure we have settled on the url structure (will confirm with @Atmosfearful) but this PR is updated and ready for another review / qa pass.

File: carbonmark/components/pages/Retire/RetireModal/index.tsx
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireModal/index.tsx:138]: Here you do not need a Go Back button.
The user can simply close the modal and see the inputs again.

The Go Back button is needed when the input form was in the same modal before.
But this is not the case here.

Please add the Figma designs again to the PR description if I am missing something here.

File: carbonmark/components/pages/Retire/RetireModal/styles.ts
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireModal/styles.ts:5]: Not in use

File: carbonmark/components/pages/Retire/RetireModal/styles.ts
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireModal/styles.ts:18]: Not in use

File: carbonmark/components/pages/Retire/RetireModal/styles.ts
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireModal/styles.ts:26]: Not in use

File: carbonmark/components/pages/Retire/RetireModal/styles.ts
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireModal/styles.ts:55]: Not in use

File: carbonmark/components/pages/Retire/RetireModal/styles.ts
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireModal/styles.ts:66]: Not in use

File: carbonmark/components/pages/Retire/RetireModal/styles.ts
Comment (ladytrekker) [inline carbonmark/components/pages/Retire/RetireModal/styles.ts:70]: Not in use ..... and so on

**Review by ladytrekker — CHANGES_REQUESTED** [review]
Thanks @psparacino for the update

I found a couple of clean-ups. Please see my comments below.

@Atmosfearful Please have a look too now and confirm the final URL structure again. See [comment](https://github.com/KlimaDAO/klimadao/pull/1145#discussion_r1210026935)

File: carbonmark/components/pages/Portfolio/AssetProject/index.tsx
Comment (psparacino) [reply carbonmark/components/pages/Portfolio/AssetProject/index.tsx:76]: Atmos had asked for this structure (different from the issue). It has changed back I can change it

File: carbonmark/components/pages/Retire/RetireForm/index.tsx
Comment (psparacino) [reply carbonmark/components/pages/Retire/RetireForm/index.tsx:70]: With the design I'm not sure how to handle the Asset Details sidebar responsiveness with only CSS? I'll take another look but I didn't see a way the first time through.

File: carbonmark/components/pages/Retire/RetireModal/index.tsx
Comment (psparacino) [reply carbonmark/components/pages/Retire/RetireModal/index.tsx:138]: this definitely makes sense but I had included the Go Back button as it was included in the design (below):

Screen Shot 2023-05-30 at 10 55 15 AM

I agree there's not really a need for it however

File: carbonmark/components/pages/Retire/RetireModal/index.tsx
Comment (psparacino) [reply carbonmark/components/pages/Retire/RetireModal/index.tsx:49]: Sounds good, I actually lifted processingRetirement up one component as well. It triggers the processing modal as soon as metamask opens instead of on submit, which I think makes sense. Catches the error in handleRetire to set it to false.

File: carbonmark/components/pages/Retire/RetireForm/index.tsx
Comment (psparacino) [reply carbonmark/components/pages/Retire/RetireForm/index.tsx:51]: Is it worth refactoring for continuity or should I just leave as is? Pretty simple form with only one validation.

**Comment by ShimonD-KlimaDAO** [comment]
Looks great! A few remaining issues.

### From my original QA pass
(1) URL ordering - this is subjective, but my impression is that we are leaning towards `/retire/address`.
(3) This explanation in the Approve tab is inaccurate:
`The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.`
Should be something like this:
`The first step is to grant the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. The next step is to execute the retirement transaction.`

This explanation in the Submit tab is inaccurate:
`The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.`
Should be something like this:
`The previous step granted the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. Your retirement has not been completed yet.`

### New issue found
The portfolio page is off-center on mobile
image

**Comment by psparacino** [comment]
> Looks great! A few remaining issues.

**Great, thanks! Comments below.**
> 
> ### From my original QA pass
> (1) URL ordering - this is subjective, but my impression is that we are leaning towards `/retire/address`. 

**will double check with Atmos**

(3) This explanation in the Approve tab is inaccurate: `The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.` Should be something like this: `The first step is to grant the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. The next step is to execute the retirement transaction.`

**Will change**
> 
> This explanation in the Submit tab is inaccurate: `The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.` Should be something like this: `The previous step granted the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. Your retirement has not been completed yet.`

**Will change. Second sentence could maybe reference approving the txn in their wallet to complete retirement**

> 
> ### New issue found
> The portfolio page is off-center on mobile image

**The Portfolio page itself I don't believe is actually part of this issue, but will see about fixing this and incorporating into this PR.**

**Comment by jabby09** [comment]
> > Looks great! A few remaining issues.
> 
> **Great, thanks! Comments below.**
> 
> > ### From my original QA pass
> > (1) URL ordering - this is subjective, but my impression is that we are leaning towards `/retire/address`.
> 
> **will double check with Atmos**
> 
> (3) This explanation in the Approve tab is inaccurate: `The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.` Should be something like this: `The first step is to grant the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. The next step is to execute the retirement transaction.`
> 
> **Will change**
> 
> > This explanation in the Submit tab is inaccurate: `The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.` Should be something like this: `The previous step granted the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. Your retirement has not been completed yet.`
> 
> **Will change. Second sentence could maybe reference approving the txn in their wallet to complete retirement**
> 
> > ### New issue found
> > The portfolio page is off-center on mobile image
> 
> **The Portfolio page itself I don't believe is actually part of this issue, but will see about fixing this and incorporating into this PR.**

@psparacino @ShimonD-KlimaDAO do not change the wording / copy for approve and submit screens. The current copy was provided specifically by @sy-zygy so I'll need to check with him first. We can proceed with what we have, if it needs to change it can be a later enhancement.

File: carbonmark/components/pages/Retire/RetireForm/index.tsx
Comment (ladytrekker) [reply carbonmark/components/pages/Retire/RetireForm/index.tsx:51]: That what just for your information.

File: carbonmark/components/pages/Portfolio/AssetProject/index.tsx
Comment (ladytrekker) [reply carbonmark/components/pages/Portfolio/AssetProject/index.tsx:76]: Ok, let's wait for @Atmosfearful here

**Review by ladytrekker — APPROVED** [review]
Thanks again @psparacino 🙇 

This PR needs to be rebased with staging again (see conflicting files)
Just rebase, accept the translation from staging and after that extract the translation on your branch again.

@Atmosfearful Please have a quick glance here too
and let's finish the decision on the URL structure.

Also keep in mind that this PR will change the styles of the Transaction modal for all transactions.

**Review by Atmosfearful — APPROVED** [review]

**Review by Atmosfearful — APPROVED** [review]

File: carbonmark/components/pages/Portfolio/AssetProject/index.tsx
Comment (Atmosfearful) [reply carbonmark/components/pages/Portfolio/AssetProject/index.tsx:76]: https://discord.com/channels/841390338324824096/964698508751949824/1113557982203084800

File: carbonmark/components/pages/Portfolio/AssetProject/index.tsx
Comment (Atmosfearful) [inline carbonmark/components/pages/Portfolio/AssetProject/index.tsx:75]: let's make sure to remove this before merging. Also, reminder when we do wire up the button, we need to pass next.js link so we can do a soft (client side) navigation, to avoid having to remount the entire app (which is causing the Login screen to reappear, which is related to this issue I've logged -> https://github.com/KlimaDAO/klimadao/issues/1174)

**Review by Atmosfearful — APPROVED** [review]
Looks good to me!
Just need to rebase.

Two follow up tickets:
https://github.com/KlimaDAO/klimadao/issues/1175
https://github.com/KlimaDAO/klimadao/issues/1174

**Review by jabby09 — APPROVED** [review]

**Pull request merged**
